src/SessionAccount.ts

```diff
@@ -86,6 +93,30 @@ export class SessionAccount {
     )
   }
 
+  private async signMessage(
+    outsideExecutionTypedData: TypedData,
+  ): Promise<Signature> {
+    const calls = (outsideExecutionTypedData.message as any).Calls.map(
+      (call: any) => ({
+        contractAddress: call.To,
+        entrypoint: call.Selector,
+        calldata: call.Calldata,
+      }),
+    )
+
+    return signOutsideExecution({
+      session: this.session,
+      sessionKey: this.sessionKey,
+      outsideExecutionTypedData,
+      argentSessionServiceUrl: this.argentSessionServiceUrl,
+      calls,
+      network:
+        this.session.chainId === constants.StarknetChainId.SN_SEPOLIA
+          ? "sepolia"
+          : "mainnet",
```

Comment:
what about other chains? maybe better to throw an error than assume mainnet

Reply:
so, for now, I'll just check that needs to be sepolia or mainnet
if not any one of these 2, throwing an error